File: backend/app/schemas/snakemake.py
from typing import Optional
from pydantic import BaseModel


class SnakemakeWrapperOut(BaseModel):
    id: str
    tool: str
    subcommand: Optional[str]
    name: Optional[str]
    description: Optional[str]
    input_names: Optional[list[str]]
    output_names: Optional[list[str]]
    category: str
    model_config = {"from_attributes": True}


class SnakemakeWorkflowOut(BaseModel):
    id: str
    name: str
    description: Optional[str]
    topics: Optional[list[str]]
    html_url: str
    stars: int
    model_config = {"from_attributes": True}


class SnakemakeCatalogStatus(BaseModel):
    wrappers: int
    workflows: int
    ready: bool

File: frontend/src/types/snakemake.ts
export interface SnakemakeWrapper {
  id: string;
  tool: string;
  subcommand: string | null;
  name: string | null;
  description: string | null;
  input_names: string[] | null;
  output_names: string[] | null;
  category: string;
}

export interface SnakemakeWorkflow {
  id: string;
  name: string;
  description: string | null;
  topics: string[] | null;
  html_url: string;
  stars: number;
}

export interface SnakemakeCatalogStatus {
  wrappers: number;
  workflows: number;
  ready: boolean;
}

export interface SnakemakeCatalogCategory {
  category: string;
  count: number;
}
